core/base/lDistanceMatrix/LDistanceMatrix.h

```diff
@@ -25,37 +25,47 @@ namespace ttk {
       DistanceType = val;
     }
 
-    template <typename T>
-    int execute(std::vector<std::vector<double>> &output,
-                const std::vector<const T *> &inputs,
+    template <typename TIn, typename TOut>
+    int execute(std::vector<TOut *> &output,
+                const std::vector<const TIn *> &inputs,
+                const size_t nPoints) const;
+    template <typename TIn, typename TOut>
+    int execute(std::vector<std::vector<TOut>> &output,
```

Comment:
please add some doxgen documentation (in particular `\warning`) to warn developers that there is a non-negligible memory overhead when using the second version of that function. In particular, the second version is still allocating a vector of `std::vector<TOut *>` of size `output.size()`.

Reply:
I understand the request, but I wanted to discuss it a little before. output is a 2D array, so if I have n points, it is of size n^2, so 4n^2 or 8n^2 bytes (float vs double).
I am allocating a new vector of size n, hence 4n or 8n size (depending if 4-byte pointers or 8-byte ones).
If n <= 1000, then the overhead is at most 8kB, which is negligible to me. If n >= 1000, the new vector is at most 1/1000  or 1/500 of the size of the original output vector, hence negligible.

If that being said you maintain your remark, no problems, I'll add it.

Reply:
thanks for adding the warning still (this is not a zero-allocation function)